src/BUILD.gn

```diff
@@ -124,6 +124,12 @@ if (chip_build_tests) {
       tests += [ "${chip_root}/src/tracing/tests" ]
     }
 
+    if (matter_enable_tracing_support &&
+        matter_trace_config ==
+        "${chip_root}/src/tracing/esp32_diagnostic_trace") {
```

Comment:
Is this needed? testable code should be generic. Can we split things into generic code (that likely does not need esp32-specific includes and then can be tested indepently) and esp-specific code (which likely has no unit tests)?